torch/csrc/jit/codegen/cuda/scheduler/mma_utils.cpp

```diff
@@ -0,0 +1,362 @@
+}
+
+void scheduleSmemCacheWrite(
+    TensorView* smem_tv,
+    const GemmTileParams& gemm_params) {
+  smem_tv->merge(-2);
+
+  int num_warp_y = gemm_params.cta_tile_shape.m / gemm_params.warp_tile_shape.m;
+  int num_warp_x = gemm_params.cta_tile_shape.n / gemm_params.warp_tile_shape.n;
+
+  // Only 128b loads are supported currently, and wouldn't need other
+  //  ones soon since this is sharedmem io.
+  int vec_word = 8;
+
+  // Create vectorized load schedule
+  smem_tv->split(-1, num_warp_y * num_warp_x * C10_WARP_SIZE * vec_word);
+  smem_tv->split(-1, num_warp_x * C10_WARP_SIZE * vec_word);
+  smem_tv->split(-1, vec_word);
+
+  smem_tv->setMemoryType(MemoryType::Shared);
+}
+
+void parallelizeSmemCacheWrite(
+    TensorView* smem_tv,
+    TensorView* global_load_reg_tv,
+    const GemmTileParams& gemm_params) {
+  smem_tv->axis(-3)->parallelize(ParallelType::TIDy);
+  smem_tv->axis(-2)->parallelize(ParallelType::TIDx);
+  smem_tv->axis(-1)->parallelize(ParallelType::Vectorize);
+  global_load_reg_tv->axis(-3)->parallelize(ParallelType::TIDy);
+  global_load_reg_tv->axis(-2)->parallelize(ParallelType::TIDx);
+
+  smem_tv->axis(0)->parallelize(ParallelType::BIDy);
+  smem_tv->axis(1)->parallelize(ParallelType::BIDx);
+  global_load_reg_tv->axis(0)->parallelize(ParallelType::BIDy);
+  global_load_reg_tv->axis(1)->parallelize(ParallelType::BIDx);
+}
+
+void parallelizeAccumulatorWrite(TensorView* tv) {
+  tv->axis(0)->parallelize(ParallelType::BIDy);
+  tv->axis(1)->parallelize(ParallelType::BIDx);
+  tv->axis(2)->parallelize(ParallelType::TIDy);
+  tv->axis(3)->parallelize(ParallelType::TIDx);
+}
+
+void parallelizeMutiplyAccumulate(TensorView* tv) {
+  tv->axis(0)->parallelize(ParallelType::BIDy);
+  tv->axis(1)->parallelize(ParallelType::BIDx);
+  // axis 2 would be the main loop
+  tv->axis(3)->parallelize(ParallelType::TIDy);
+  tv->axis(4)->parallelize(ParallelType::TIDx);
+}
+
+void parallelizeSmemWrite(TensorView* tv) {
+  tv->axis(0)->parallelize(ParallelType::BIDy);
+  tv->axis(1)->parallelize(ParallelType::BIDx);
+}
+
+void orderCtaTileAxes(TensorView* tv) {
+  auto tv_domain = tv->domain()->domain();
+  TORCH_INTERNAL_ASSERT(tv_domain.size() == 6);
+  auto tv_root = TensorDomain::noReductions(tv->getMaybeRFactorDomain());
+
+  std::unordered_map<IterDomain*, int> root_order;
+  for (auto root_idx : c10::irange(tv_root.size())) {
+    root_order[tv_root[root_idx]] = root_idx;
+  }
+
+  std::unordered_map<int, int> mnk_order;
+  for (auto idx : c10::irange(3, 6)) {
+    auto id_def = tv_domain[idx]->definition();
+    TORCH_INTERNAL_ASSERT(id_def);
+    auto id_def_in = id_def->input(0);
+    auto root_idx_it = root_order.find(id_def_in->as<IterDomain>());
+    TORCH_INTERNAL_ASSERT(root_idx_it != root_order.end());
+    mnk_order[idx] = root_idx_it->second + 3;
+  }
+
+  tv->reorder(mnk_order);
+}
+
+} // namespace
+
+void scheduleGemm(Fusion* fusion, const GemmTileParams& gemm_params) {
```

Comment:
I will probably rename this to `scheduleCutlassStyleGemm`.  This is just meant to be an example and maybe later divided into a few useful utilities.

User should be welcome to custome write everything in this function, except that there will be a few restrictions with using mma. The checks will be built out in a separate PR.